### PR TITLE
Fix insertMany failpoint

### PR DESCRIPTION
**Pull request by ShaneHarvey**
With `times: 2` the retry attempt for this insert still succeeds:
```
> db.adminCommand( { "configureFailPoint" : "onPrimaryTransactionalWrite", "mode" : { "times" : 2 }, "data" : { "failBeforeCommitExceptionCode" : 1 } })
{ "ok" : 1, "operationTime" : Timestamp(1509648432, 5) }
> var txnCmd = {"insert":"test","documents":[{_id:1},{_id:2},{_id:3}], "lsid": {"id": BinData(4, "nGmSpxKoTcWrYclQh5xF0A==")}, txnNumber: NumberLong(12), ordered: true}
> db.runCommand(txnCmd);
2017-11-02T14:47:25.664-0400 E QUERY    [thread1] Error: error doing query: failed: network error while attempting to run command 'insert' on host '127.0.0.1:27017'  :
DB.prototype.runCommand@src/mongo/shell/db.js:132:1
@(shell):1:1
2017-11-02T14:47:25.666-0400 I NETWORK  [thread1] trying reconnect to 127.0.0.1:27017 (127.0.0.1) failed
2017-11-02T14:47:25.667-0400 I NETWORK  [thread1] reconnect 127.0.0.1:27017 (127.0.0.1) ok
> db.runCommand(txnCmd);
{
	"n" : 3,
	"opTime" : {
		"ts" : Timestamp(1509648446, 3),
		"t" : NumberLong(1)
	},
	"electionId" : ObjectId("7fffffff0000000000000001"),
	"ok" : 1,
	"operationTime" : Timestamp(1509648446, 3)
}
```
With "alwaysOn" the retry fails like we want:
```
> db.adminCommand( { "configureFailPoint" : "onPrimaryTransactionalWrite", "mode" : "alwaysOn", "data" : { "failBeforeCommitExceptionCode" : 1 } })
{ "ok" : 1, "operationTime" : Timestamp(1509648549, 5) }
> var txnCmd = {"insert":"test","documents":[{_id:1},{_id:2},{_id:3}], "lsid": {"id": BinData(4, "nGmSpxKoTcWrYclQh5xF0A==")}, txnNumber: NumberLong(12), ordered: true}
> db.runCommand(txnCmd);
2017-11-02T14:49:29.482-0400 E QUERY    [thread1] Error: error doing query: failed: network error while attempting to run command 'insert' on host '127.0.0.1:27017'  :
DB.prototype.runCommand@src/mongo/shell/db.js:132:1
@(shell):1:1
2017-11-02T14:49:29.483-0400 I NETWORK  [thread1] trying reconnect to 127.0.0.1:27017 (127.0.0.1) failed
2017-11-02T14:49:29.485-0400 I NETWORK  [thread1] reconnect 127.0.0.1:27017 (127.0.0.1) ok
> db.runCommand(txnCmd);
2017-11-02T14:49:30.529-0400 E QUERY    [thread1] Error: error doing query: failed: network error while attempting to run command 'insert' on host '127.0.0.1:27017'  :
DB.prototype.runCommand@src/mongo/shell/db.js:132:1
@(shell):1:1
2017-11-02T14:49:30.532-0400 I NETWORK  [thread1] trying reconnect to 127.0.0.1:27017 (127.0.0.1) failed
2017-11-02T14:49:30.533-0400 I NETWORK  [thread1] reconnect 127.0.0.1:27017 (127.0.0.1) ok
```

### CODE REVIEW

File: source/retryable-writes/tests/insertMany.yml [hunk 1]
```diff
@@ -45,12 +45,12 @@ tests:
     -
         description: "InsertMany fails after multiple network errors"
         failPoint:
-            mode: { times: 2 }
+            mode: "alwaysOn"
```
Comment (jmikola) [inline source/retryable-writes/tests/insertMany.yml]: I won't argue that this fixes the test, but do we know why? If `failBeforeCommitExceptionCode` is set, the write should never be committed, so I would expect 2 fail point triggers would be enough to prevent the retryable write from succeeding.

Comment (ShaneHarvey) [reply source/retryable-writes/tests/insertMany.yml]: Sorry, I do not know why. I'll ask Kal.

Comment (jmikola) [reply source/retryable-writes/tests/insertMany.yml]: Quoting @jsmulrow from Slack:

> So it looks like if the server isn’t able to insert the documents as a batch, it will try to insert each document one at a time: https://github.com/mongodb/mongo/blob/f96e0ef10baac1d2ca254c2c4ef78ef0ff8a0542/src/mongo/db/ops/write_ops_exec.cpp#L387-L388
> So my guess is that the batched attempt fails, then the server tries each document individually, which exhausts the failpoint. So then when the command is retried, the failpoint is inactive.

In this case, we may be blowing through `times=2` in the first write command. `times=3` may be the conservative fix. That said, while I see value in crafting the test to exercise the boundary condition, one could argue that is more appropriate for the server's test suite and we should stick with the most resilient option of "alwaysOn".

Comment (ShaneHarvey) [reply source/retryable-writes/tests/insertMany.yml]: I stuck with "alwaysOn" and added a comment.
